.github/workflows/ci.yml

```diff
@@ -182,28 +193,44 @@ jobs:
           SIGNING_KEY: ${{ secrets.SIGNING_KEY }}
         run: .github/ci.sh sign $NAME-with-solvers.tar.gz
 
```

Comment:
Let's add some comments here describing what the intent is.  I'm not even sure myself, but I think it's probably something like:

```
 ##################################################################
 # We upload an archive containing saw, and also and archive containing saw and the set of possible SMT solvers,
 # but only for our "primary" distribution (currently: ghc 8.10.7).  These archives are utilized in subsequent CI jobs,
 # but are also published for external users, and  are therefore signed.
 #
 # In addition, if we built with HPC, we upload the hpc-enabled binaries as a separate package along with a
 # parallel package containing the HPC-specific files generated during the build process.  These are mostly used
 # for subsequent CI jobs that will generate a coverage report, and the binaries are essentially the same as those
 # collected for the previous operation, but they are captured in their original cabal-generated locations where
 # they are expected to live for subsequent coverage collection.
```

Some of my following comments are based on my interpretation stated here, so mis-interpretation here may affect the validity of the following comments.

Reply:
Your interpretation is correct. I added comments better explaining what is going on